src/silx/gui/dialog/ColormapDialog.py

```diff
@@ -1723,9 +1736,28 @@ def _autoscaleModeUpdated(self):
 
         colormap = self.getColormap()
         if colormap is not None:
+            activate_saturation = self._autoScaleCombo.currentText() == "Percentile"
             with self._colormapChange:
+                if activate_saturation:
+                    colormap.setSaturation(self._saturation.value())
+                else:
+                    colormap.setSaturation(Colormap.DEFAULT_SATURATION)
                 colormap.setAutoscaleMode(mode)
 
+        self._updateSaturationVisibility()
+        self._updateWidgetRange()
+
+    def _updateSaturationVisibility(self):
+        activate_saturation = self._autoScaleCombo.currentText() == "Percentile"
+        self._saturation.setVisible(activate_saturation)
+        self._saturationLabel.setVisible(activate_saturation)
```

Comment:
IMO best to enable/disable rather than show/hide widgets.
BTW, you can use `setEnabled` on the layout, so that can be controlled once for both the label and slide.

Reply:
:ok_hand: ;  for enabling / disabling from the layout it was not working on my side. I guess I am missing something byut what...